

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>bioseq package &mdash; bioseq 0.1 documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="../_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
        <script type="text/javascript" src="../_static/jquery.js"></script>
        <script type="text/javascript" src="../_static/underscore.js"></script>
        <script type="text/javascript" src="../_static/doctools.js"></script>
        <script type="text/javascript" src="../_static/language_data.js"></script>
    
    <script type="text/javascript" src="../_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../index.html" class="icon icon-home"> bioseq
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <!-- Local TOC -->
              <div class="local-toc"><ul>
<li><a class="reference internal" href="#">bioseq package</a><ul>
<li><a class="reference internal" href="#submodules">Submodules</a></li>
<li><a class="reference internal" href="#module-bioseq.bioseq">bioseq module</a></li>
<li><a class="reference internal" href="#module-bioseq.dnaseq">dnaseq module</a></li>
<li><a class="reference internal" href="#module-bioseq.rnaseq">rnaseq module</a></li>
<li><a class="reference internal" href="#module-bioseq.proteinseq">proteinseq module</a></li>
<li><a class="reference internal" href="#module-bioseq.utils">utils module</a></li>
<li><a class="reference internal" href="#module-bioseq">Module contents</a></li>
</ul>
</li>
</ul>
</div>
            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">bioseq</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../index.html">Docs</a> &raquo;</li>
        
      <li>bioseq package</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="../_sources/source/bioseq.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="bioseq-package">
<h1>bioseq package<a class="headerlink" href="#bioseq-package" title="Permalink to this headline">¶</a></h1>
<div class="section" id="submodules">
<h2>Submodules<a class="headerlink" href="#submodules" title="Permalink to this headline">¶</a></h2>
</div>
<div class="section" id="module-bioseq.bioseq">
<span id="bioseq-module"></span><h2>bioseq module<a class="headerlink" href="#module-bioseq.bioseq" title="Permalink to this headline">¶</a></h2>
<dl class="class">
<dt id="bioseq.bioseq.BioSeq">
<em class="property">class </em><code class="descclassname">bioseq.bioseq.</code><code class="descname">BioSeq</code><span class="sig-paren">(</span><em>sequence</em>, <em>seq_type</em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/bioseq/bioseq.html#BioSeq"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#bioseq.bioseq.BioSeq" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<p>General purpose class for handling and manipulating biological sequences and perform operations on them.</p>
<dl class="method">
<dt id="bioseq.bioseq.BioSeq._assert_seq_type">
<code class="descname">_assert_seq_type</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="../_modules/bioseq/bioseq.html#BioSeq._assert_seq_type"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#bioseq.bioseq.BioSeq._assert_seq_type" title="Permalink to this definition">¶</a></dt>
<dd><p>Assert a proper valid has been chosen for seq_type</p>
</dd></dl>

<dl class="method">
<dt id="bioseq.bioseq.BioSeq._assert_valid_sequence">
<code class="descname">_assert_valid_sequence</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="../_modules/bioseq/bioseq.html#BioSeq._assert_valid_sequence"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#bioseq.bioseq.BioSeq._assert_valid_sequence" title="Permalink to this definition">¶</a></dt>
<dd><p>assert that all the tokens in the sequence are valid for that sequence type</p>
</dd></dl>

<dl class="method">
<dt id="bioseq.bioseq.BioSeq.frequency">
<code class="descname">frequency</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="../_modules/bioseq/bioseq.html#BioSeq.frequency"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#bioseq.bioseq.BioSeq.frequency" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculates the relative frequency of each token in the sequence</p>
</dd></dl>

<dl class="method">
<dt id="bioseq.bioseq.BioSeq.gc_content">
<code class="descname">gc_content</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="../_modules/bioseq/bioseq.html#BioSeq.gc_content"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#bioseq.bioseq.BioSeq.gc_content" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate the gc_content of the sequence</p>
</dd></dl>

<dl class="method">
<dt id="bioseq.bioseq.BioSeq.load">
<code class="descname">load</code><span class="sig-paren">(</span><em>filename</em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/bioseq/bioseq.html#BioSeq.load"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#bioseq.bioseq.BioSeq.load" title="Permalink to this definition">¶</a></dt>
<dd><p>Load object from file</p>
</dd></dl>

<dl class="method">
<dt id="bioseq.bioseq.BioSeq.pretty_print">
<code class="descname">pretty_print</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="../_modules/bioseq/bioseq.html#BioSeq.pretty_print"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#bioseq.bioseq.BioSeq.pretty_print" title="Permalink to this definition">¶</a></dt>
<dd><p>Outputs a prettified and informative string onto the console, describing the current sequence</p>
</dd></dl>

<dl class="attribute">
<dt id="bioseq.bioseq.BioSeq.reverse">
<code class="descname">reverse</code><em class="property"> = {}</em><a class="headerlink" href="#bioseq.bioseq.BioSeq.reverse" title="Permalink to this definition">¶</a></dt>
<dd><p>rules for reverse_complement</p>
</dd></dl>

<dl class="method">
<dt id="bioseq.bioseq.BioSeq.reverse_complement">
<code class="descname">reverse_complement</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="../_modules/bioseq/bioseq.html#BioSeq.reverse_complement"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#bioseq.bioseq.BioSeq.reverse_complement" title="Permalink to this definition">¶</a></dt>
<dd><p>Given a Biological sequence, generate the reverse complement sequence for the specific type. Uses the static variable reverse</p>
</dd></dl>

<dl class="method">
<dt id="bioseq.bioseq.BioSeq.save">
<code class="descname">save</code><span class="sig-paren">(</span><em>filename</em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/bioseq/bioseq.html#BioSeq.save"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#bioseq.bioseq.BioSeq.save" title="Permalink to this definition">¶</a></dt>
<dd><p>Save an object to a file</p>
</dd></dl>

<dl class="attribute">
<dt id="bioseq.bioseq.BioSeq.seq_type">
<code class="descname">seq_type</code><em class="property"> = None</em><a class="headerlink" href="#bioseq.bioseq.BioSeq.seq_type" title="Permalink to this definition">¶</a></dt>
<dd><p>DNA, RNA, PROTEIN</p>
</dd></dl>

<dl class="attribute">
<dt id="bioseq.bioseq.BioSeq.sequence">
<code class="descname">sequence</code><em class="property"> = None</em><a class="headerlink" href="#bioseq.bioseq.BioSeq.sequence" title="Permalink to this definition">¶</a></dt>
<dd><p>The current sequence</p>
</dd></dl>

<dl class="attribute">
<dt id="bioseq.bioseq.BioSeq.valid_tokens">
<code class="descname">valid_tokens</code><em class="property"> = {}</em><a class="headerlink" href="#bioseq.bioseq.BioSeq.valid_tokens" title="Permalink to this definition">¶</a></dt>
<dd><p>Which tokens a given biological sequence can have</p>
</dd></dl>

<dl class="attribute">
<dt id="bioseq.bioseq.BioSeq.valid_types">
<code class="descname">valid_types</code><em class="property"> = {'DNA', 'PROTEIN', 'RNA'}</em><a class="headerlink" href="#bioseq.bioseq.BioSeq.valid_types" title="Permalink to this definition">¶</a></dt>
<dd><p>What are the valid values for seq_type</p>
</dd></dl>

</dd></dl>

</div>
<div class="section" id="module-bioseq.dnaseq">
<span id="dnaseq-module"></span><h2>dnaseq module<a class="headerlink" href="#module-bioseq.dnaseq" title="Permalink to this headline">¶</a></h2>
<dl class="class">
<dt id="bioseq.dnaseq.DNASeq">
<em class="property">class </em><code class="descclassname">bioseq.dnaseq.</code><code class="descname">DNASeq</code><span class="sig-paren">(</span><em>sequence</em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/bioseq/dnaseq.html#DNASeq"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#bioseq.dnaseq.DNASeq" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#bioseq.bioseq.BioSeq" title="bioseq.bioseq.BioSeq"><code class="xref py py-class docutils literal notranslate"><span class="pre">bioseq.bioseq.BioSeq</span></code></a></p>
<p>Biological Sequence specific for DNA</p>
<dl class="method">
<dt id="bioseq.dnaseq.DNASeq.codon_usage">
<code class="descname">codon_usage</code><span class="sig-paren">(</span><em>aminoacid</em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/bioseq/dnaseq.html#DNASeq.codon_usage"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#bioseq.dnaseq.DNASeq.codon_usage" title="Permalink to this definition">¶</a></dt>
<dd><p>Given a sequence,  transcript it to RNA and call the respective codon_usage to get the frequency of each codon in it that maps to a specific aminoacid. Returns a dict.</p>
</dd></dl>

<dl class="method">
<dt id="bioseq.dnaseq.DNASeq.open_reading_frames">
<code class="descname">open_reading_frames</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="../_modules/bioseq/dnaseq.html#DNASeq.open_reading_frames"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#bioseq.dnaseq.DNASeq.open_reading_frames" title="Permalink to this definition">¶</a></dt>
<dd><p>Given a sequence, get all the possible open reading frames for each reading frame after transcripting to RNASeq</p>
</dd></dl>

<dl class="attribute">
<dt id="bioseq.dnaseq.DNASeq.reverse">
<code class="descname">reverse</code><em class="property"> = {'A': 'T', 'C': 'G', 'G': 'C', 'T': 'A'}</em><a class="headerlink" href="#bioseq.dnaseq.DNASeq.reverse" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="bioseq.dnaseq.DNASeq.transcription">
<code class="descname">transcription</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="../_modules/bioseq/dnaseq.html#DNASeq.transcription"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#bioseq.dnaseq.DNASeq.transcription" title="Permalink to this definition">¶</a></dt>
<dd><p>Performs a transcription of the original sequence from DNA to RNA and returns a RNASeq object</p>
</dd></dl>

<dl class="method">
<dt id="bioseq.dnaseq.DNASeq.translation">
<code class="descname">translation</code><span class="sig-paren">(</span><em>start=0</em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/bioseq/dnaseq.html#DNASeq.translation"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#bioseq.dnaseq.DNASeq.translation" title="Permalink to this definition">¶</a></dt>
<dd><p>Given a DNA sequence, transcript it to RNA and call the respective translation method for RNA sequences. Starts codon split at position <strong>start</strong> (default is 0). Returns a ProteinSeq instance.</p>
</dd></dl>

<dl class="attribute">
<dt id="bioseq.dnaseq.DNASeq.valid_tokens">
<code class="descname">valid_tokens</code><em class="property"> = {'A', 'C', 'G', 'T'}</em><a class="headerlink" href="#bioseq.dnaseq.DNASeq.valid_tokens" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

</div>
<div class="section" id="module-bioseq.rnaseq">
<span id="rnaseq-module"></span><h2>rnaseq module<a class="headerlink" href="#module-bioseq.rnaseq" title="Permalink to this headline">¶</a></h2>
<dl class="class">
<dt id="bioseq.rnaseq.RNASeq">
<em class="property">class </em><code class="descclassname">bioseq.rnaseq.</code><code class="descname">RNASeq</code><span class="sig-paren">(</span><em>sequence</em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/bioseq/rnaseq.html#RNASeq"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#bioseq.rnaseq.RNASeq" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#bioseq.bioseq.BioSeq" title="bioseq.bioseq.BioSeq"><code class="xref py py-class docutils literal notranslate"><span class="pre">bioseq.bioseq.BioSeq</span></code></a></p>
<p>Biological Sequence specific for RNA</p>
<dl class="method">
<dt id="bioseq.rnaseq.RNASeq._reading_frames">
<code class="descname">_reading_frames</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="../_modules/bioseq/rnaseq.html#RNASeq._reading_frames"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#bioseq.rnaseq.RNASeq._reading_frames" title="Permalink to this definition">¶</a></dt>
<dd><p>Given a sequence, get all the possible aminoacid sequences (always 6)</p>
</dd></dl>

<dl class="method">
<dt id="bioseq.rnaseq.RNASeq.codon_usage">
<code class="descname">codon_usage</code><span class="sig-paren">(</span><em>aminoacid</em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/bioseq/rnaseq.html#RNASeq.codon_usage"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#bioseq.rnaseq.RNASeq.codon_usage" title="Permalink to this definition">¶</a></dt>
<dd><p>Given a sequence, calculate the frequency of each codon in it that maps to a specific aminoacid. Returns a dict.</p>
</dd></dl>

<dl class="attribute">
<dt id="bioseq.rnaseq.RNASeq.dict_of_aminoacids">
<code class="descname">dict_of_aminoacids</code><em class="property"> = {'AAA': 'K', 'AAC': 'N', 'AAG': 'K', 'AAU': 'N', 'ACA': 'T', 'ACC': 'T', 'ACG': 'T', 'ACU': 'T', 'AGA': 'R', 'AGC': 'S', 'AGG': 'R', 'AGU': 'S', 'AUA': 'I', 'AUC': 'I', 'AUG': 'M', 'AUU': 'I', 'CAA': 'Q', 'CAC': 'H', 'CAG': 'Q', 'CAU': 'H', 'CCA': 'P', 'CCC': 'P', 'CCG': 'P', 'CCU': 'P', 'CGA': 'R', 'CGC': 'R', 'CGG': 'R', 'CGU': 'R', 'CUA': 'L', 'CUC': 'L', 'CUG': 'L', 'CUU': 'L', 'GAA': 'E', 'GAC': 'D', 'GAG': 'E', 'GAU': 'D', 'GCA': 'A', 'GCC': 'A', 'GCG': 'A', 'GCU': 'A', 'GGA': 'G', 'GGC': 'G', 'GGG': 'G', 'GGU': 'G', 'GUA': 'V', 'GUC': 'V', 'GUG': 'V', 'GUU': 'V', 'UAA': '_', 'UAC': 'Y', 'UAG': '_', 'UAU': 'Y', 'UCA': 'S', 'UCC': 'S', 'UCG': 'S', 'UCU': 'S', 'UGA': '_', 'UGC': 'C', 'UGG': 'W', 'UGU': 'C', 'UUA': 'L', 'UUC': 'F', 'UUG': 'L', 'UUU': 'F'}</em><a class="headerlink" href="#bioseq.rnaseq.RNASeq.dict_of_aminoacids" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="bioseq.rnaseq.RNASeq.open_reading_frames">
<code class="descname">open_reading_frames</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="../_modules/bioseq/rnaseq.html#RNASeq.open_reading_frames"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#bioseq.rnaseq.RNASeq.open_reading_frames" title="Permalink to this definition">¶</a></dt>
<dd><p>Given a sequence, get all the possible open reading frames for each reading frame</p>
</dd></dl>

<dl class="attribute">
<dt id="bioseq.rnaseq.RNASeq.reverse">
<code class="descname">reverse</code><em class="property"> = {'A': 'U', 'C': 'G', 'G': 'C', 'U': 'A'}</em><a class="headerlink" href="#bioseq.rnaseq.RNASeq.reverse" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="bioseq.rnaseq.RNASeq.translation">
<code class="descname">translation</code><span class="sig-paren">(</span><em>start=0</em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/bioseq/rnaseq.html#RNASeq.translation"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#bioseq.rnaseq.RNASeq.translation" title="Permalink to this definition">¶</a></dt>
<dd><p>Given an RNA sequence, translate its 3-token codons into the respective aminoacids. Starts codon split at position <strong>start</strong> (default is 0). Returns a ProteinSeq instance.</p>
</dd></dl>

<dl class="attribute">
<dt id="bioseq.rnaseq.RNASeq.valid_tokens">
<code class="descname">valid_tokens</code><em class="property"> = {'A', 'C', 'G', 'U'}</em><a class="headerlink" href="#bioseq.rnaseq.RNASeq.valid_tokens" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

</div>
<div class="section" id="module-bioseq.proteinseq">
<span id="proteinseq-module"></span><h2>proteinseq module<a class="headerlink" href="#module-bioseq.proteinseq" title="Permalink to this headline">¶</a></h2>
<dl class="class">
<dt id="bioseq.proteinseq.ProteinSeq">
<em class="property">class </em><code class="descclassname">bioseq.proteinseq.</code><code class="descname">ProteinSeq</code><span class="sig-paren">(</span><em>sequence</em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/bioseq/proteinseq.html#ProteinSeq"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#bioseq.proteinseq.ProteinSeq" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#bioseq.bioseq.BioSeq" title="bioseq.bioseq.BioSeq"><code class="xref py py-class docutils literal notranslate"><span class="pre">bioseq.bioseq.BioSeq</span></code></a></p>
<p>Biological Sequence specific for PROTEINS</p>
<dl class="method">
<dt id="bioseq.proteinseq.ProteinSeq.gc_content">
<code class="descname">gc_content</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="../_modules/bioseq/proteinseq.html#ProteinSeq.gc_content"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#bioseq.proteinseq.ProteinSeq.gc_content" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate the gc_content of the sequence</p>
</dd></dl>

<dl class="method">
<dt id="bioseq.proteinseq.ProteinSeq.reverse_complement">
<code class="descname">reverse_complement</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="../_modules/bioseq/proteinseq.html#ProteinSeq.reverse_complement"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#bioseq.proteinseq.ProteinSeq.reverse_complement" title="Permalink to this definition">¶</a></dt>
<dd><p>Given a Biological sequence, generate the reverse complement sequence for the specific type. Uses the static variable reverse</p>
</dd></dl>

<dl class="attribute">
<dt id="bioseq.proteinseq.ProteinSeq.valid_tokens">
<code class="descname">valid_tokens</code><em class="property"> = {'A', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'K', 'L', 'M', 'N', 'P', 'Q', 'R', 'S', 'T', 'V', 'W', 'Y', '_'}</em><a class="headerlink" href="#bioseq.proteinseq.ProteinSeq.valid_tokens" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

</div>
<div class="section" id="module-bioseq.utils">
<span id="utils-module"></span><h2>utils module<a class="headerlink" href="#module-bioseq.utils" title="Permalink to this headline">¶</a></h2>
<dl class="function">
<dt id="bioseq.utils.get_dict_of_dna_to_aminoacids">
<code class="descclassname">bioseq.utils.</code><code class="descname">get_dict_of_dna_to_aminoacids</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="../_modules/bioseq/utils.html#get_dict_of_dna_to_aminoacids"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#bioseq.utils.get_dict_of_dna_to_aminoacids" title="Permalink to this definition">¶</a></dt>
<dd><p>Get a dictionary that maps RNA codons onto their respective proteins</p>
</dd></dl>

<dl class="function">
<dt id="bioseq.utils.read_fasta">
<code class="descclassname">bioseq.utils.</code><code class="descname">read_fasta</code><span class="sig-paren">(</span><em>filename</em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/bioseq/utils.html#read_fasta"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#bioseq.utils.read_fasta" title="Permalink to this definition">¶</a></dt>
<dd><p>Function that reads a FASTA file into a list of its ProteinSeq entries. Returns list of tuples (comment, ProteinSeq)</p>
</dd></dl>

</div>
<div class="section" id="module-bioseq">
<span id="module-contents"></span><h2>Module contents<a class="headerlink" href="#module-bioseq" title="Permalink to this headline">¶</a></h2>
</div>
</div>


           </div>
           
          </div>
          <footer>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2019, Miguel Sozinho Ramalho

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>